pkgs/servers/sql/postgresql/default.nix

```diff
@@ -35,18 +43,34 @@ let
 
     hardeningEnable = lib.optionals (!stdenv.cc.isClang) [ "pie" ];
 
-    outputs = [ "out" "lib" "doc" "man" ];
-    setOutputFlags = false; # $out retains configureFlags :-/
+    outputs = [ "out" "dev" "lib" "doc" "man" ];
 
-    buildInputs =
-      [ zlib readline openssl libxml2 ]
-      ++ lib.optionals icuEnabled [ icu ]
+    buildInputs = [
+      zlib
+      readline
+      openssl
+      libxml2
+      icu
+    ]
       ++ lib.optionals lz4Enabled [ lz4 ]
       ++ lib.optionals enableSystemd [ systemd ]
       ++ lib.optionals gssSupport [ libkrb5 ]
       ++ lib.optionals (!stdenv.isDarwin) [ libossp_uuid ];
 
-    nativeBuildInputs = [ makeWrapper ] ++ lib.optionals icuEnabled [ pkg-config ];
+    nativeBuildInputs = [
+      makeWrapper
+      pkg-config
+      # We are patching configure.ac
+      autoreconfHook269
+      perl
+      (if atLeast "13" then docbook_xml_dtd_45 else docbook_xml_dtd_42)
+      libxslt
+      docbook-xsl-nons
+    ] ++ lib.optionals (!atLeast "11") [
+      opensp # for osx
+      # Postgres 10 only migrated man pages to XML, not the rest of the docs.
+      docbook_sgml_dtd_42
```

Comment:
```suggestion
```
we already have this package a few lines above.

Reply:
That is the schema for XML variant of DocBook, this one is for SGML. As the comment says, the old version needs both.

Reply:
oh, they are not the same. 😅